CMakeLists.txt

```diff
@@ -1,10 +1,275 @@
+    )
+
+    set_target_properties(${target} PROPERTIES
+        MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL"
+        VS_GLOBAL_CLToolChain "v143"
+        VS_GLOBAL_PlatformToolset "v143"
+        VS_GLOBAL_DisableSpecificWarnings "MSB8027;MSB8065;4244;4267;4996;4273;4141;4005;4141" 
+        VS_GLOBAL_TreatWarningAsError "false"
+        VS_GLOBAL_WarningLevel "0"
+    )
+
+    configure_cpu_optimized_unity_build(${target})
+endfunction()
+
+# Linux detection function
+function(configure_linux_target target)
+    if(WIN32)
+        return()
+    endif()
+
+    target_compile_options(${target} PRIVATE
+        -Wall
+        -Wextra
+        -Wpedantic
+        -Wno-unknown-pragmas
+        -fPIC
+        $<$<OR:$<CONFIG:Release>,$<CONFIG:Distribution>>:-O3>
+    )
+
+    if(WARNINGS_AS_ERRORS)
+        target_compile_options(${target} PRIVATE -Werror)
+    endif()
+
+    set_target_properties(${target} PROPERTIES
+        POSITION_INDEPENDENT_CODE ON
+        CXX_VISIBILITY_PRESET hidden
+        VISIBILITY_INLINES_HIDDEN ON
+    )
+endfunction()
+
+# Common target configuration function
+function(configure_target_build target)
+    if(WIN32 AND MSVC)
+        configure_windows_target(${target})
+        
+        target_include_directories(${target} BEFORE PRIVATE 
+            ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES}
+            ${PROJECT_BINARY_DIR}
+            ${PROJECT_SOURCE_DIR}/third_party/JoltPhysics
+        )
+    else()
+        configure_linux_target(${target})
+    endif()
+
+    target_compile_definitions(${target} PRIVATE
+        JPH_PROFILE_ENABLED=0
+        JPH_DEBUG_RENDERER=0
+        JPH_FLOATING_POINT_EXCEPTIONS_ENABLED=0
+        JPH_USE_SSE4_1=1
+        JPH_USE_SSE4_2=1
+        JPH_TRACK_BROADPHASE_STATS=0
+        JPH_STATISTICS=0
+        JPH_DISABLE_CUSTOM_ALLOCATOR
+    )
+endfunction()
+
+# Source files print function
+function(print_source_files target)
+    if(MSVC)
+        get_target_property(target_sources ${target} SOURCES)
+        if(target_sources)
+            set(cpp_files "")
+            foreach(source ${target_sources})
+                if(source MATCHES ".*\\.(cpp|cxx|cc)$")
+                    get_filename_component(filename ${source} NAME)
+                    list(APPEND cpp_files ${filename})
+                endif()
+            endforeach()
+            
+            list(LENGTH cpp_files total_files)
+            if(total_files GREATER 0)
+                list(SORT cpp_files)
+                message(STATUS "Files for ${target}:")
+                foreach(filename ${cpp_files})
+                    message(STATUS "  ${filename}")
+                endforeach()
+            endif()
+        endif()
+    endif()
+endfunction()
+
+# OS detection
+if(WIN32)
+    set(THRIVE_OS "windows")
+    set(THRIVE_ARCH "x64")
+else()
+    set(THRIVE_OS "linux")
+    execute_process(COMMAND uname -m OUTPUT_VARIABLE THRIVE_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
```

Comment:
Did you add support for compiling as 32-bit? If you didn't I think this line is totally useless.

Reply:
i did pal